.github/workflows/npm-update.yml

```diff
@@ -0,0 +1,39 @@
+name: npm-update
+on:
+  # weekly automatic update on early Tuesday morning (before release day)
+  schedule:
+    - cron: '50 2 * * 2'
+  # can be run manually on https://github.com/cockpit-project/cockpit-ostree/actions
+  workflow_dispatch:
+jobs:
+  npm-update:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up dependencies
+        run: sudo apt-get install -y npm make
+
+      - name: Set up configuration and secrets
+        run: |
+          echo '${{ secrets.GITHUB_TOKEN }}' > ~/.config/github-token
+
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Run bot
+        run: |
+          make bots
+          # we do the PR ourselves, much easier and avoids the need for a custom token
+          bots/npm-update --verbose --dry
+          # compute the changed module from git diff; that's a bit awkward, let's perhaps add an npm-update option?
+          echo -n 'MODULE=' >> $GITHUB_ENV
+          git diff package.json | sed -n '/^+ / { s/^[^"]*"//; s/".*$//; p; q }' >> $GITHUB_ENV
+
+      # this silently skips creating the PR if there are no changes, and updates an existing PR
+      - name: Create/update Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: "package.json: Update ${{ env.MODULE }} package dependency"
+          title: "[bot] package.json: Update ${{ env.MODULE }} package dependency"
+          branch: bots/npm-update
+          delete-branch: true
```

Comment:
The update happens every Tuesday. If we don't close/merge PR in one week, this will become problematic. can we just use generated branch name (at least `npm-update-<date>`)?

Reply:
Problematic how? I specifically chose to *not* pile up lots of branch names. The workflow will update an already existing PR in the following week if it is still open. I find that much nicer than having potentially duplicated/outdated PRs.

Reply:
You start with master, create commit and then try to push into branch that already has commit in it. It may cause conflicts (and likely will as they might update the same file). Or am I misunderstanding what this does?

Reply:
According to the [documentation](https://github.com/peter-evans/create-pull-request/blob/master/docs/concepts-guidelines.md#how-the-action-works), it force-pushes on changes, and it always starts with the base branch (current master).

If we want to, we can switch to a [timestamp suffix](https://github.com/marketplace/actions/create-pull-request#alternative-strategy---always-create-a-new-pull-request-branch) just like we have with cockpituous now, if we don't like the behaviour. The docs explicitly don't recommend that, though. But I'd actually like to try and see how the default behaviour works for us, and if we don't like it, we can still change the behaviour?

Reply:
I'd be ok then with seeing the default behaviour, especially if it's recommended in the docs.